app/controllers/publishers/omniauth_callbacks_controller.rb

```diff
@@ -1,10 +1,14 @@
 module Publishers
   class OmniauthCallbacksController < Devise::OmniauthCallbacksController
 
+    include PublishersHelper
+
     # ToDo: rework
-    before_action :require_publisher
+    before_action :require_publisher, only: [:register_youtube_channel]
+    before_action :require_publisher_created_through_youtube_auth,
+                  only: %i(register_youtube_channel)
```

Comment:
should this be `only: %i(youtube_login)`

Reply:
I missed the word `not` in the method_name.